common/maplib/map.mjs

```diff
@@ -7,7 +7,159 @@
  * Each cell of the map is usually a string or boolean.
  * The documentation for format of the map is found at the end of this file.
  */
+
+
+/*
+ * The map coordinate:
+ * Be careful not to get confused with "canvas coordinate".
+ * notation: (y, x)
+ * unit: grid (or cell, in other words)
+ *
+ *           ...
+ *
+ *   (2,0)   (2,1)   (2,2)
+ *
+ *   (1,0)   (1,1)   (1,2)   ...
+ *
+ *   (0,0)   (0,1)   (0,2)
+ */
+class MapCoord {
```

Comment:
[Optional] Consider moving MapCoord into another .mjs for better compatibility with import().

Reply:
I think that it is reasonable to place this class in map.mjs, since it is a class related to map.
Besides, this module uses named export to export this class and uses default export for `GameMap`. This is for compatibility with the existing code.